docs/standard/native-interop/calling-conventions.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Unmanaged calling conventions
+description: Learn how to specify unmanaged calling conventions in .NET
+author: jkotas
+ms.date: 08/19/2023
+---
+
+# Unmanaged calling conventions
+
+[Calling conventions](https://en.wikipedia.org/wiki/Calling_convention) desribe low-level details for how method arguments and return values
+are passed between the caller and the called method.
+
+It is important that the unmanaged calling convention declared in P/Invoke declaration matches the unmanaged calling convention
+used by the native implementation. Unmanaged calling convention mismatches lead to data corruptions and fatal crashes that require
+low-level debugging skills to diagnose.
+
+## Platform default calling convention
+
+Most platforms use one canonical calling conventions and explicitly specified calling convention is unnecessary in most cases.
+
+For x86 architecture, the default calling convention is platform-specific. StdCall is the default calling convention on Windows x86
+and it is used by most Win32 APIs. Cdecl is the default calling convention on Linux x86. Windows ports of open source libraries that
+originated on Unix often use Cdecl calling convention even on Windows x86 and it is necessary to explicitly specifify the Cdecl calling
+convention in P/Invoke declarations for interop with these libraries.
+
+For non-x86 architectures, both StdCall and Cdecl calling conventions are treated as the canonical platform default calling convention.
+
+## Specifying calling conventions in managed P/Invoke declarations
```

Comment:
Do we want to mention the `CallingConvention.Winapi` option?

Reply:
`CallingConvention.Winapi` is no-op. It is same as default; people do not really need to know that it exists. I think it is enough for it to be mentioned in https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.callingconvention?view=net-7.0#fields . 

I plan to add link from CallingConvention.Winapi docs to "Platform default calling convention" paragraph in this doc as a follow up.